rasterio/_warp.pyx

```diff
@@ -733,28 +758,54 @@ cdef GDALDatasetH auto_create_warped_vrt(
     double dfMaxError,
     const GDALWarpOptions *psOptions,
     const char **transformer_options,
-) nogil:
+):
+    """Makes a best-fit WarpedVRT around a dataset.
 
-    IF (CTE_GDAL_MAJOR_VERSION, CTE_GDAL_MINOR_VERSION) >= (3, 2):
-        return GDALAutoCreateWarpedVRTEx(
-            hSrcDS,
-            pszSrcWKT,
-            pszDstWKT,
-            eResampleAlg,
-            dfMaxError,
-            psOptions,
-            transformer_options,
-        )
+    Returns
+    -------
+    GDALDatasetH
+        Owned by the caller.
 
+    """
+    cdef GDALDatasetH hds_warped = NULL
+
+    IF (CTE_GDAL_MAJOR_VERSION, CTE_GDAL_MINOR_VERSION) >= (3, 2):
+        try:
+            with nogil:
+                hds_warped = GDALAutoCreateWarpedVRTEx(
+                    hSrcDS,
+                    pszSrcWKT,
+                    pszDstWKT,
+                    eResampleAlg,
+                    dfMaxError,
+                    psOptions,
+                    transformer_options,
+                )
+            _ = exc_wrap(0)
+        except CPLE_AppDefinedError as err:
+            if hds_warped != NULL:
+                log.info("Treating error as warning: err=%r", err)
```

Comment:
@vincentsarago here's the fix for the VRT case.

Reply:
🙏 confirmed it work!

do you want me to add a test for this?